test/test_configuration.py

```diff
@@ -112,23 +115,108 @@ def test_descriptions(self):
             parameters = {"para": QueryParameter(name="sort", description=3)}
             self.UrlApiConfig(_create_endpoints(parameters=parameters))
 
+    def test_good_example_with_choices(self):
+        parameters = {"para": QueryParameter(name="sort",
+                                             example=3,
+                                             choices=[1,2,3])}
+        self.UrlApiConfig(_create_endpoints(parameters=parameters))
+
+    def test_bad_example_with_choices(self):
+        with self.assertRaises(RestClientConfigurationError) as exc:
+
+            parameters = {"para": QueryParameter(name="sort",
+                                                 example=3,
+                                                 choices=[1,2,4])}
+            self.UrlApiConfig(_create_endpoints(parameters=parameters))
+
+        self.assertEqual(exc.exception.args[0], "example should be one of the choices")
```

Comment:
Kudos for checking that the right `RestClientConfigurationError` has been raised.